tests/python/gpu/test_operator_gpu.py

```diff
@@ -2121,6 +2121,69 @@ def test_context_num_gpus():
     # Test that num_gpus reports at least one GPU, as the test is run on a GPU host.
     assert mx.context.num_gpus() > 0
 
+def math_log(shape, dtype, check_value):
+    np_x = np.random.rand(shape[0], shape[1])
+    x = mx.nd.array(np_x, dtype=dtype)
+    mx.nd.waitall()
+    y = mx.nd.log(data=x)
+    y.wait_to_read()
+    if check_value:
+        x_ = x.as_in_context(mx.cpu())
+        mx.nd.waitall()
+        y_ = mx.nd.log(data=x_)
+        y_.wait_to_read()
+        assert_almost_equal(y.asnumpy(), y_.asnumpy())
+
+def math_erf(shape, dtype, check_value):
+    np_x = np.random.rand(shape[0], shape[1])
+    x = mx.nd.array(np_x, dtype=dtype)
+    mx.nd.waitall()
+    y = mx.nd.erf(data=x)
+    y.wait_to_read()
+    if check_value:
+        x_ = x.as_in_context(mx.cpu())
+        mx.nd.waitall()
+        y_ = mx.nd.erf(data=x_)
+        y_.wait_to_read()
+        assert_almost_equal(y.asnumpy(), y_.asnumpy())
+
+def math_square(shape, dtype, check_value):
+    np_x = np.random.rand(shape[0], shape[1])
+    x = mx.nd.array(np_x, dtype=dtype)
+    mx.nd.waitall()
```

Comment:
Why do we need to frequently add waitall and wait_to_read in tests?

Reply:
The test case is initially to calcuatee the time cost of the calculations on different size tensors, wwwill remove the waitall, wait_to_read fron the test case as the test case targting the correctness of the calculation.